sdk/typescript/src/signers/txn-data-serializers/local-txn-data-serializer.ts

```diff
@@ -352,6 +356,123 @@ export class LocalTxnDataSerializer implements TxnDataSerializer {
     serialized.set(dataBytes, TYPE_TAG.length);
     return new Base64DataBuffer(serialized);
   }
+
+  /**
+   * Deserialize BCS encoded bytes into `SignableTransaction`
+   */
+  public async deserializeTransactionBytesToSignableTransaction(
+    bytes: Base64DataBuffer
+  ): Promise<
+    UnserializedSignableTransaction | UnserializedSignableTransaction[]
+  > {
+    return this.transformTransactionDataToSignableTransaction(
+      await this.deserializeTransactionBytesToTransactionData(bytes)
+    );
+  }
+
+  /**
+   * Deserialize BCS encoded bytes into `TransactionData`
+   */
+  public async deserializeTransactionBytesToTransactionData(
+    bytes: Base64DataBuffer
+  ): Promise<TransactionData> {
+    const version = await this.provider.getRpcApiVersion();
+    const format = shouldUseOldSharedObjectAPI(version)
+      ? 'TransactionData_Deprecated'
+      : 'TransactionData';
+    return bcs.de(format, bytes.getData().slice(TYPE_TAG.length));
+  }
+
+  /**
+   * Deserialize `TransactionData` to `SignableTransaction`
+   */
+  public async transformTransactionDataToSignableTransaction(
+    tx: TransactionData
+  ): Promise<
+    UnserializedSignableTransaction | UnserializedSignableTransaction[]
+  > {
+    if ('Single' in tx.kind) {
+      return this.transformTransactionToSignableTransaction(
+        tx.kind.Single,
+        tx.gasBudget,
+        tx.gasPayment
+      );
+    }
+    return Promise.all(
+      tx.kind.Batch.map((t) =>
+        this.transformTransactionToSignableTransaction(
+          t,
+          tx.gasBudget,
+          tx.gasPayment
+        )
+      )
+    );
+  }
+
+  public async transformTransactionToSignableTransaction(
+    tx: Transaction,
+    gasBudget: number,
+    gasPayment?: SuiObjectRef
+  ): Promise<UnserializedSignableTransaction> {
+    if ('Pay' in tx) {
+      return {
+        kind: 'pay',
+        data: {
+          inputCoins: tx.Pay.coins.map((c) => c.objectId),
+          recipients: tx.Pay.recipients,
+          amounts: tx.Pay.amounts,
+          gasPayment: gasPayment?.objectId,
+          gasBudget,
+        },
+      };
+    } else if ('Call' in tx) {
+      return {
+        kind: 'moveCall',
+        data: {
+          packageObjectId: tx.Call.package.objectId,
+          module: tx.Call.module,
+          function: tx.Call.function,
+          typeArguments: tx.Call.typeArguments,
+          arguments: await new CallArgSerializer(
```

Comment:
I wonder if this should be an option in the event that you just want to partially deserialize calls.

Reply:
In that case they can use `deserializeTransactionBytesToTransactionData` on line 376